### DISCUSSION TITLE
How to solve the problem that error establishing server connection: client disconnected

### DISCUSSION THREAD

**Discussion by chunxifeng** [Melledy/Grasscutter#1361] [Support]
Hello everyone, I have logged in successfully, but not every time I can log in, and when I can't log in the mitmproxy log shows the following：
> 127.0.0.1:1422: error establishing server connection: client disconnected
127.0.0.1:1422: CONNECT log-upload-os.hoyoverse.com:443
             << 502 Bad Gateway 70b
127.0.0.1:1424: error establishing server connection: client disconnected
127.0.0.1:1424: CONNECT api-account-os.hoyoverse.com:443
             << 502 Bad Gateway 71b
127.0.0.1:1426: client connect
127.0.0.1:1426: client disconnect
127.0.0.1:1426: error establishing server connection: client disconnected
127.0.0.1:1426: CONNECT log-upload-os.hoyoverse.com:443
             << 502 Bad Gateway 70b
127.0.0.1:1428: client connect
127.0.0.1:1428: client disconnect
127.0.0.1:1428: error establishing server connection: client disconnected
127.0.0.1:1428: CONNECT log-upload-os.hoyoverse.com:443
             << 502 Bad Gateway 70b

And Grasscutter shows
> [Dispatch] Client 127.0.0.1 request: query_region_list

There is no output after this. I've tried everything I could find in the last few days and still can't solve it, hope for a solution, thanks!

**Comment by IFNEF** [comment]
free your 443 port

**Reply by chunxifeng** [reply]
Thanks, but the problem still exists, I tried other ports as well and could log in successfully, now I started trying different proxies

**Reply by JamesYuanzzz99** [reply]
I also encountered a problem with "query_region_list" and did not appear on the Grasscutter login page. How did you resolve it at that time? Can you elaborate on it?

**Reply by chunxifeng** [reply]
> I also encountered a problem with "query_region_list" and did not appear on the Grasscutter login page. How did you resolve it at that time? Can you elaborate on it?

I used another proxy and its connection is here:[https://github.com/Coooookies/OceanLauncher](url)

**Comment by chunxifeng** [comment]
Now I tried every proxy method I could find, most of them only logged me in occasionally, until I found [OceanLauncher](https://github.com/Coooookies/OceanLauncher), which allowed me to log in 100%! The other proxy methods don't work as described by everyone, so the problem isn't actually solved, which is probably because I triggered some qualifying bug? I'm not sure how OceanLauncher differs from the other proxies, but on my computer it works perfectly with Grasscutter.
